src/ngraph_builder.cc

```diff
@@ -785,12 +817,19 @@ static Status TranslateBatchMatMulOp(
         "should have the same size");
   }
   if (n_dims == 2) {
-    SaveNgOp(ng_op_map, op->name(),
-             make_shared<ngraph::op::Dot>(ng_lhs, ng_rhs));
+    auto dot_output = make_shared<ngraph::op::Dot>(ng_lhs, ng_rhs);
+    if (!got_first_node) {
+      SaveNgOp(first_ng_op_map, op->name(), dot_output);
+    }
+    SaveNgOp(ng_op_map, op->name(), dot_output);
+
   } else {
     auto output_shape = ng_lhs_shape;
     output_shape[n_dims - 1] = ng_rhs_shape[1];
     auto dot_output = make_shared<ngraph::op::Dot>(ng_lhs, ng_rhs);
+    if (!got_first_node) {
+      SaveNgOp(first_ng_op_map, op->name(), dot_output);
+    }
```

Comment:
Q - Why do we have a boolean here to check if it is first node or not unlike other ops?

Can't we just do this for Saving the map in line #822, is there a reason in assigning it to dot_output?
```
SaveNgOp(first_ng_op_map, op->name(),
             make_shared<ngraph::op::Dot>(ng_lhs, ng_rhs));
```

Reply:
The boolean is to check if ng_lhs and ng_rhs (the input nodes) have been modified as they might be in few cases (when tf_adjx is true) etc. If they have been modified they are the entry nodes.

The boolean makes the check simpler otherwise we need to check for many conditions

Reply:
hmm, ok. thanks for the explanation